File: factorial_algorithm.py
#This is the factorial algorithm in python
def factorial(n):
    if n<=1:
        return 1
    return n*factorial(n-1)

number=int(input("Enter a positive number: "))
if number<0:
    number=int(input("Enter a positive number: "))
    print("The factorial of ",number,"is ",factorial(number))
else:
    print("The factorial of ",number,"is ",factorial(number))

File: Factorial_algorithm.c
//This is a program to find the factorial of a number in C

#include <stdio.h>
int factorial(int a){
    if(a==0||a==1){
        return 1;
    }
    return a * factorial(a-1);
}
int main(){
    int n;
    printf("Enter a positive number of your choice: ");
    scanf("%d",&n);

    if (n<0){
        printf("Enter a positive number: ");
    }
    else{
        printf("%d is the factorial of %d",factorial(n),n);
    }
    return 0;
}